add git commit hash in cargo-contract version

atenjin opened the pull request.

cmichi reviewed (changes requested):
@atenjin The commit hash is missing in the second command here:
```
$ cargo run -- --version
cargo-contract 0.8.0-bd0db50-x86_64-linux-gnu
$ cargo run -- contract --version
cargo-contract 0.8.0
```

Besides that I think it's fine the way you copied the `generate_cargo_keys` stuff, avoiding polluting the codebase with substrate-named environment variables. So you can remove the comments regarding that.

cmichi commented on build.rs:
```diff
@@ -15,18 +15,23 @@
 };
 
 use anyhow::Result;
+use platforms::*;
```
I would prefer not having a wildcard import here.
```suggestion
use platforms::{TARGET_ARCH, TARGET_ENV, TARGET_OS};
```

atenjin replied:
have fixed

atenjin commented:
> @atenjin The commit hash is missing in the second command here:
> 
> ```
> $ cargo run -- --version
> cargo-contract 0.8.0-bd0db50-x86_64-linux-gnu
> $ cargo run -- contract --version
> cargo-contract 0.8.0
> ```
> 
> Besides that I think it's fine the way you copied the `generate_cargo_keys` stuff, avoiding polluting the codebase with substrate-named environment variables. So you can remove the comments regarding that.

thanks, I find out the place to add version for this tip. 

Now I have removed the todo command and added suitable version hint.

cmichi reviewed (approved):
Thanks @atenjin !
